xds/internal/balancer/clusterresolver/config.go

```diff
@@ -19,110 +19,44 @@ package clusterresolver
-// and Fallbackspolicy are post-processed, and for each, the first installed
-// policy is kept.
-type edsConfigJSON struct {
-	ChildPolicy                []*loadBalancingConfig
-	FallbackPolicy             []*loadBalancingConfig
-	ClusterName                string
-	EDSServiceName             string
-	MaxConcurrentRequests      *uint32
-	LRSLoadReportingServerName *string
-}
-
-// UnmarshalJSON parses the JSON-encoded byte slice in data and stores it in l.
-// When unmarshalling, we iterate through the childPolicy/fallbackPolicy lists
-// and select the first LB policy which has been registered.
-func (l *EDSConfig) UnmarshalJSON(data []byte) error {
-	var configJSON edsConfigJSON
-	if err := json.Unmarshal(data, &configJSON); err != nil {
-		return err
-	}
+	// Must have at least one element. Results from each discovery mechanism are
+	// concatenated together in successive priorities.
+	DiscoveryMechanisms []balancerconfig.DiscoveryMechanism `json:"discoveryMechanisms,omitempty"`
 
-	l.ClusterName = configJSON.ClusterName
-	l.EDSServiceName = configJSON.EDSServiceName
-	l.MaxConcurrentRequests = configJSON.MaxConcurrentRequests
-	l.LrsLoadReportingServerName = configJSON.LRSLoadReportingServerName
-
-	for _, lbcfg := range configJSON.ChildPolicy {
-		if balancer.Get(lbcfg.Name) != nil {
-			l.ChildPolicy = lbcfg
-			break
-		}
-	}
-
-	for _, lbcfg := range configJSON.FallbackPolicy {
-		if balancer.Get(lbcfg.Name) != nil {
-			l.FallBackPolicy = lbcfg
-			break
-		}
-	}
-	return nil
-}
-
-// MarshalJSON returns a JSON encoding of l.
-func (l *EDSConfig) MarshalJSON() ([]byte, error) {
-	return nil, fmt.Errorf("EDSConfig.MarshalJSON() is unimplemented")
-}
-
-// loadBalancingConfig represents a single load balancing config,
-// stored in JSON format.
-type loadBalancingConfig struct {
-	Name   string
-	Config json.RawMessage
-}
+	// LocalityPickingPolicy is policy for locality picking.
+	//
+	// This policy's config is expected to be in the format used by the
+	// weighted_target policy.  Note that the config should include an empty
+	// value for the "targets" field; that empty value will be replaced by one
+	// that is dynamically generated based on the EDS data. Optional; defaults
+	// to "weighted_target".
+	LocalityPickingPolicy *internalserviceconfig.BalancerConfig `json:"localityPickingPolicy,omitempty"`
+
+	// EndpointPickingPolicy is policy for endpoint picking.
+	//
+	// This will be configured as the policy for each child in the
+	// locality-policy's config. Optional; defaults to "round_robin".
+	EndpointPickingPolicy *internalserviceconfig.BalancerConfig `json:"endpointPickingPolicy,omitempty"`
 
-// MarshalJSON returns a JSON encoding of l.
-func (l *loadBalancingConfig) MarshalJSON() ([]byte, error) {
-	return nil, fmt.Errorf("loadBalancingConfig.MarshalJSON() is unimplemented")
+	// FIXME: read and warn if endpoint is not roundrobin or locality is not
+	// weightedtarget.
```

Comment:
Do it now? Or s/FIXME/TODO/?

Reply:
Changed to TODO.
Actually, we will never do this. In the next PR, both these two fields will be deleted.